src/buildercore/threadbare/common.py

```diff
@@ -0,0 +1,114 @@
+import os
+from functools import reduce
+
+
+class PromptedException(BaseException):
+    pass
+
+
+def first(x):
+    "returns the first element in an collection of things"
+    if x is None:
+        return x
+    try:
+        if isinstance(x, dict):
+            return list(x.items())[0]
```

Comment:
this is suspect. native ordered dicts were only introduced in python3.7 and the docker containers are using python3.5

Reply:
support for dicts removed